EIPS/eip-5994.md

```diff
@@ -0,0 +1,85 @@
+discussions-to: https://ethereum-magicians.org/t/eip-6000-token-pods-extensions-erc20-erc721/11782
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-11-18
+requires: 20, 721
+---
+
+## Abstract
+
+Token Pods concept introduces token extensions for the most popular token standards: ERC20, ERC721 and ERC1155, which allows holders of assets to connect their balances to Pods, which could track changes of their balances.
+
+## Motivation
+
+This Token Pods concept allows to have permissionless extensions for tokens, which can be used for incentivisations (aka farms) or delegations or any other ideas.
+
+Example implementation can be found here:
+- https://github.com/1inch/erc20-pods
+
+Examples of Pods:
+- https://github.com/1inch/farming
+- https://github.com/1inch/delegating
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+interface IPod {
+    function updateBalances(address from, address to, uint256 amount) external;
+}
+
+interface IERC20Pods is IERC20 {
+    function hasPod(address account, address pod) external view returns(bool);
+    function podsCount(address account) external view returns(uint256);
+    function podAt(address account, uint256 index) external view returns(address);
+    function pods(address account) external view returns(address[] memory);
+    function podBalanceOf(address pod, address account) external view returns(uint256);
+
+    function addPod(address pod) external;
+    function removePod(address pod) external;
+    function removeAllPods() external;
+}
+
+interface IERC721Pods is IERC721 {
+    function hasPod(address account, address pod) external view returns(bool);
+    function podsCount(address account) external view returns(uint256);
+    function podAt(address account, uint256 index) external view returns(address);
+    function pods(address account) external view returns(address[] memory);
+    function podBalanceOf(address pod, address account) external view returns(uint256);
+
+    function addPod(address pod) external;
+    function removePod(address pod) external;
+    function removeAllPods() external;
+}
+```
+
+### Pod specification
+
+- Pod MUST implement interface `IPod` and fit execution of `updateBalances` method in 200,000 gas and never revert.
+
+- Pod SHOULD make sure `updateBalances(...)` method was triggered by `token` it was designated for.
+
+- Pod can track changes of balances of users who voluntarily added this pod, all other account will be represented as `address(0)` and interactions with such accounts could be considered as mints and burns.
+
+- Pod SHOULD NOT try to change balances of the token it was designated for, since it would introduce reentrancy, which will not pass.
+
+## Rationale
+
+## Backwards Compatibility
+
+Fully backward compatibile with ERC20 and ERC721.
```

Comment:
```suggestion
Fully backward compatible with EIP-20, EIP-721, and EIP-1155.
```
There was a spelling mistake, and EIP-1155 was missing. Also, as previously mentioned, the format for EIPs is currently `EIP-X`, where `X` is the EIP's number.

(This might change. If it does, I'll change it myself)